### PR TITLE
feat: removed kafka and added queue management

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,9 +4,15 @@ WORKDIR /app
...
+RUN apk update \
+    && apk add --no-cache gcc git python3-dev musl-dev linux-headers \
+    libc-dev  rsync zsh \
+    findutils wget util-linux grep libxml2-dev libxslt-dev \
+    &&  pip3 install --upgrade pip
+RUN pip install --upgrade pip
 COPY requirements.txt ./
 RUN pip3 install -r ./requirements.txt -t /app
-RUN apk del py3-pip
+# RUN apk del py3-pip
```
Comment (asafMasa) [inline Dockerfile]: check

File: confd/production.tmpl [hunk 2]
```diff
@@ -1,16 +1,10 @@
...
-        "ssl_certfile": {{ if ne (getv "/configuration/exporter/worker/kafka/ssl/certfile" "") "" }}"{{ getv "/configuration/exporter/worker/kafka/ssl/certfile" }}"{{ else }}{{ "null" }}{{ end }},
-        "ssl_keyfile": {{ if ne (getv "/configuration/exporter/worker/kafka/ssl/keyfile" "") "" }}"{{ getv "/configuration/exporter/worker/kafka/ssl/keyfile" }}"{{ else }}{{ "null" }}{{ end }},
-        "ssl_password": {{ if ne (getv "/configuration/exporter/worker/kafka/ssl/password" "") "" }}"{{ getv "/configuration/exporter/worker/kafka/ssl/password" }}"{{ else }}{{ "null" }}{{ end }}
+    "server": {
+        "job_type": "{{ getv "/configuration/exporter/worker/server/job/type" "raster" }}",
+        "task_type": "{{ getv "/configuration/exporter/worker/server/task/type" "exporter" }}",
+        "job_manager_url": {{ getv "/configuration/exporter/worker/server/job/manager/url" "http://127.0.0.1:8082" }},
+        "heartbeat_manager_url": {{ getv "/configuration/exporter/server/heartbeat/manager/url" "http://127.0.0.1:8081" }},
+        "heartbeat_interval_ms": {{ getv "/configuration/exporter/worker/max/attempts" "0.2" }}
```
Comment (asafMasa) [inline confd/production.tmpl]: heartbeat_interval_ms -> /max/attempts ?????!!!

File: confd/production.toml [hunk 3]
```diff
@@ -2,17 +2,11 @@
...
-    "/configuration/exporter/worker/kafka/poll_records",
-    "/configuration/exporter/worker/kafka/ssl/context",
-    "/configuration/exporter/worker/kafka/ssl/cafile",
-    "/configuration/exporter/worker/kafka/ssl/certfile",
-    "/configuration/exporter/worker/kafka/ssl/keyfile",
-    "/configuration/exporter/worker/kafka/ssl/password",
+    "/configuration/exporter/worker/server/task_type",
+    "/configuration/exporter/worker/server/job_type",
+    "/configuration/exporter/worker/server/job_manager_url",
+    "/configuration/exporter/worker/server/heartbeat_manager_url",
```
Comment (asafMasa) [inline confd/production.toml]: heartbeat/manager/url in the tmpl file... 
check all fields

File: src/exportImage.py [hunk 4]
```diff
@@ -71,79 +76,86 @@ def __init__(self):
...
+                            except Exception as e:
+                                self.log.info(
+                                    f'Failed uploading task "{task_id}" to s3.')
+                                raise e
+                            finally:
+                                # Delete local files
+                                self.delete_local_directory(directory_name)
+                                await self.queue_handler.ack(job_id, task_id)
+                        else:
+                            await asyncio.sleep(2.0)
```
Comment (asafMasa) [inline src/exportImage.py]: talk with me with again on this issue

File: src/exportImage.py [hunk 5]
```diff
@@ -71,79 +76,86 @@ def __init__(self):
...
     def warp_progress_callback(self, complete, message, unknown):
         """
         Callback for updating export progress according to geopackage creation process.
         """
         # Calculate warp percent of the whole process
+
         percent = floor(complete * self.warp_percent)
-        self.__helper.save_update(
-            unknown["taskId"], Status.IN_PROGRESS.value, unknown["filename"], percent)
+        asyncio.run_coroutine_threadsafe(
```
Comment (asafMasa) [inline src/exportImage.py:143]: coroutine?

File: src/exportImage.py [hunk 6]
```diff
@@ -71,79 +76,86 @@ def __init__(self):
...
+            self.queue_handler.update_progress(unknown['job_id'], unknown['task_id'], percent), loop=self.loop)
+
 
     def overviews_progress_callback(self, complete, message, unknown):
         """
         Callback for updating export progress according to geopackage overview build.
         """
         # Calculate build overview percent of the whole process
+        ## TODO Ask Shimon why overview is not working
```
Comment (asafMasa) [inline src/exportImage.py]: ?

File: src/exportImage.py [hunk 7]
```diff
@@ -71,79 +76,86 @@ def __init__(self):
...
         percent = floor(complete * (self.overview_percent))
+        asyncio.run_coroutine_threadsafe(
+            self.queue_handler.update_progress(unknown['job_id'], unknown['task_id'], self.warp_percent + percent), self.loop)
         # Final percent = warp percent + build overviews percent
-        self.__helper.save_update(
-            unknown["taskId"], Status.IN_PROGRESS.value, unknown["filename"], self.warp_percent + percent)
 
-    def upload_to_s3(self, filename, directoryName, output_format, full_path):
+    def upload_to_s3(self, filename, directory_name, output_format, full_path):
```
Comment (asafMasa) [inline src/exportImage.py]: filename -> file_name

File: src/handler.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        self.__config = read_json(config_path)
+        self.queue_handler = TaskHandler(self.__config["server"]["job_type"], self.__config["server"]["task_type"],
+                                         self.__config["server"]["job_manager_url"], self.__config["server"]["heartbeat_manager_url"],
+                                         self.__config["server"]["heartbeat_interval_ms"], self.log)
+        self.__exportImage = ExportImage(self.queue_handler, self.loop)
+
+    async def handle_tasks(self):
+        try:
+            while True:
+                task = await self.queue_handler.dequeue(5)
```
Comment (asafMasa) [inline src/handler.py]: 5? no hard coded values - put in a static class member

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: Dockerfile (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply Dockerfile]: fixed in 0444dfa

File: confd/production.tmpl (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply confd/production.tmpl]: fixed in 66594f8

File: confd/production.toml (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply confd/production.toml]: fixed in 66594f8

File: src/exportImage.py (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 3ce17b5

File: src/exportImage.py (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in e398c7b

File: src/handler.py (hunk 8 above)
Comment (CL-SHLOMIKONCHA) [reply src/handler.py]: fixed in a2ede34

**Review by asafMasa — APPROVED** [review]

File: src/exportImage.py (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: moved into config value

**Review by asafMasa — APPROVED** [review]

**Pull request merged**